eclipse-solargraph-plugin/src/main/java/io/github/pyvesb/eclipse_solargraph/launch/RubyRunDelegate.java

```diff
@@ -0,0 +1,155 @@
+	}
+
+	private String[] getFileCommand(String fileName, String arguments) {
+		StringBuilder ruby = new StringBuilder();
+		if (!SYSTEM_RUBY.getValue()) {
+			ruby.append(RUBY_DIR.getValue());
+			if (ruby.length() > 0) {
+				ruby.append(File.separator);
+			}
+		}
+		ruby.append("ruby");
+		List<String> command = new ArrayList<>();
+		command.add(ruby.toString());
+		command.add(fileName);
+		command.addAll(Arrays.asList(arguments.split(" ")));
+		return command.toArray(new String[0]);
+	}
+
+	@Override
+	public void launch(ISelection selection, String mode) {
+		if (selection instanceof IStructuredSelection) {
+			@SuppressWarnings("unchecked")
+			Iterator<Object> selectionIterator = ((IStructuredSelection) selection).iterator();
+			while (selectionIterator.hasNext()) {
+				Object element = selectionIterator.next();
+				IResource resource = null;
+				if (element instanceof IResource) {
+					resource = (IResource) element;
+				} else if (element instanceof IAdaptable) {
+					resource = ((IAdaptable) element).getAdapter(IResource.class);
+				}
+
+				if (resource != null) {
+					try {
+						ILaunchConfiguration launchConfig = getLaunchConfiguration(mode, resource);
+						launchConfig.launch(mode, new NullProgressMonitor());
+					} catch (CoreException e) {
+						e.printStackTrace();
+					}
+					return;
+				}
+			}
+		}
+		Display.getDefault().asyncExec(() -> {
+			MessageDialog.openError(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(), "Error",
+					"Issue running");
+		});
+
+	}
+
+	@Override
+	public void launch(IEditorPart editor, String mode) {
+		IEditorInput input = editor.getEditorInput();
+		IFile file = input.getAdapter(IFile.class);
+
+		try {
+			ILaunchConfiguration launchConfig = getLaunchConfiguration(mode, file);
+			launchConfig.launch(mode, new NullProgressMonitor());
+		} catch (CoreException e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	private ILaunchConfiguration getLaunchConfiguration(String mode, IResource resource) {
+		ILaunchManager launchManager = DebugPlugin.getDefault().getLaunchManager();
+		ILaunchConfigurationType configType = launchManager
+				.getLaunchConfigurationType("io.github.pyvesb.eclipse_solargraph.launch.RubyRunDelegate"); //$NON-NLS-1$
+		try {
+			ILaunchConfiguration[] launchConfigurations = launchManager.getLaunchConfigurations(configType);
+
+			String configName = resource.getName();
+
+			for (ILaunchConfiguration iLaunchConfiguration : launchConfigurations) {
+				if (iLaunchConfiguration.getName().equals(configName)) {
+					return iLaunchConfiguration;
+				}
+			}
+			configName = launchManager.generateLaunchConfigurationName(configName);
+			ILaunchConfigurationWorkingCopy wc = configType.newInstance(null, configName);
+			wc.setAttribute(PROGRAM, resource.getLocation().toString());
+			wc.setAttribute(DebugPlugin.ATTR_WORKING_DIRECTORY, resource.getProject().getLocation().toString());
+			wc.doSave();
+			return wc;
+		} catch (CoreException e) {
```

Comment:
This can probably be added to the method signature and not handled here: both calling points already have a similar try/catch block. :wink:

Reply:
Done.